File: Sorting/Python/bubblesort.py
def bubblesort(x):
    if len(x) == 0:
        return
    size = len(x)
    for i in range(size):
        for j in range(size-1-i):
            if x[j] > x[j+1]:
                temp = x[j]
                x[j] = x[j+1]
                x[j+1] = temp
    return
    
def main():
    testCaseOne = [3,2,4,5,6,7,1]
    testCaseTwo = [5,4,3,2,1]

    bubblesort(testCaseOne)
    bubblesort(testCaseTwo)

    print("Testcase 1: ", testCaseOne)
    print("Testcase 2: ", testCaseTwo)

if __name__ == "__main__":
    main()

File: Sorting/C/bubblesort.c
#include <stdio.h>
#include "swap.h"

void bubblesort(int size, int ar[]){
    int i,j;
    for(i = 0; i<size; i++){
        for(j = 0; j<size-i-1; j++)
        {
            if(ar[j]>ar[j+1]){
                swap(&ar[j], &ar[j+1]);
            }
        }
    }
}

int main(){
    int size = 8, ar[] = {11,3,2,7,6,4,3,12};
    bubblesort(size, ar);
    int i;
    for(i = 0; i<size; i++)
        printf("%d ", ar[i]);

    return 0;
}
